backend/modules/model_gateway/model_gateway.py

```diff
@@ -1,85 +1,119 @@
 import json
-from pydantic import BaseModel
-from typing import List, Optional
-from backend.types import ModelConfig, ModelProviderConfig, ModelType
+import os
+from typing import List
+
 from langchain.embeddings.base import Embeddings
-from langchain_openai import OpenAIEmbeddings
 from langchain_core.language_models.chat_models import BaseChatModel
+from langchain_openai import OpenAIEmbeddings
 from langchain_openai.chat_models import ChatOpenAI
-import os
+
+from backend.logger import logger
+from backend.types import ModelConfig, ModelProviderConfig, ModelType
+
 
 class ModelGateway:
     config: List[ModelProviderConfig]
     modelsToProviderMap = {}
```

Comment:
nit: convert to snake case
naming is also pretty confusing, isn't this `model_id_to_provider_config` ?

Reply:
@innoavator: Please check on this.